### PR TITLE
SwG Button Impression Logging

### PR DESCRIPTION
**Pull request by elijahsoria**
Adds impression logging to ButtonApi on attach of SwG button. Adds tests to verify that impression logging is being done.

Note that we decided to add impression logging on our end as well as have publishers log button impression with LoggerApi because we will then be able to grab all impressions done on the SwG button and differentiate between publisher logged impressions and SwG logged impressions by looking at the event origin.

### CODE REVIEW

File: src/runtime/runtime-test.js [hunk 1]
```diff
@@ -1687,7 +1687,7 @@ describes.realWin('ConfiguredRuntime', {}, env => {
 
       const button = runtime.createButton();
       await button.click();
-      expect(count).to.equal(1);
+      expect(count).to.not.equal(0);
```
Comment (sohanirao) [inline src/runtime/runtime-test.js]: I dont understand the reason for this change. Is this carried over from a previous change or something changed in how button clicks are counted?

Comment (ChrisAntaki) [reply src/runtime/runtime-test.js]: I mentioned this too, it's been fixed 👍

File: src/runtime/button-api-test.js [hunk 2]
```diff
@@ -187,21 +188,42 @@ describes.realWin('ButtonApi', {}, env => {
...
+    let count = 0;
+    sandbox
+      .stub(ClientEventManager.prototype, 'logSwgEvent')
+      .withArgs(AnalyticsEvent.ACTION_SWG_BUTTON_CLICK, true)
+      .callsFake(() => count++);
+    await button.click();
+    expect(count).to.equal(1);
+  });
+
+  it('should log button impression on create', async () => {
```
Comment (sohanirao) [inline src/runtime/button-api-test.js]: And this is because create calls attach. yes?

Comment (elijahsoria) [reply src/runtime/button-api-test.js]: Yes that is correct. I just wanted to test both functions explicitly.

**Pull request merged**